api/scenario_api_utils/scripts/ros_time_publisher.py

```diff
@@ -1,44 +1,49 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
-
-import rospy
+import rclpy
+from rclpy.node import Node
 import time
 from rosgraph_msgs.msg import Clock
 
-
-class RosTimePublisher():
-
+class RosTimePublisher(Node):
     def __init__(self):
-        self.time_rate_ = rospy.get_param("~time_rate", 1.0)
-        # rospy does not have WallTime)
-        self.initial_time_ = rospy.Time.from_sec(time.time())
+        super().__init__('ros_time_publisher')
+        use_sim_time_param = rclpy.parameter.Parameter(
+            "use_sim_time",
+            rclpy.Parameter.Type.BOOL,
+            False)
+        self.set_parameters([use_sim_time_param])
```

Comment:
I wonder you call `set_parameters` before you declare the next one in the next line. Wouldn't it make more sense to `set_parameters[use_sim_time, time_rate]`?

Reply:
I removed use_sim_time anyways so this shouldn't matter anymore.